lib/charms/data_platform_libs/v0/data_models.py

```diff
@@ -248,10 +255,11 @@ def read(relation_data: MutableMapping[str, str], obj: Type[T]) -> T:
         **{
             field_name: (
                 relation_data[parsed_key]
-                if field.annotation in [int, str, float]
+                if field.outer_type_ in DataBagNativeTypes
                 else json.loads(relation_data[parsed_key])
             )
-            for field_name, field in obj.__fields__.items()  # pyright: ignore[reportGeneralTypeIssues]
+            for field_name, field in obj.__fields__.items()
+            # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
Do you need this?

Reply:
This is a library managed by the data platform team and not a part of my code. I'm not able to optimise it.